python/llm/src/bigdl/llm/ggml/convert_model.py

```diff
@@ -0,0 +1,40 @@
+from bigdl.llm.ggml.convert import _convert_to_ggml
+from bigdl.llm.ggml.quantize import quantize
+from pathlib import Path
+import time
+
+
+def convert_model(input_path: str,
+                  output_path: str = None,
+                  model_family: str = 'llama',
+                  dtype: str = 'int4'):
+    """
+    Convert Hugging Face llama-like / gpt-neox-like / bloom-like model to lower precision
+
+    :param input_path: str, path of model, for example `./llama-7b-hf`.
+    :param output_path: Save path of output quantized model. Default to `None`.
+            If you don't specify this parameter, quantized model will be saved in
+            the same directory as the input and just replace precision with quantize_type
+            like `./ggml-model-q4_0.bin`.
+    :param model_family: Which model family your input model belongs to. Default to `llama`.
+            Now only `llama`/`bloomz`/`gptneox` are supported.
+    :param dtype: Which quantized precision will be converted
+            Now only int4 supported.
+    """
+
```

Comment:
you need to check if the input values are valid

Reply:
* The `input_path` has checked in `_convert_to_ggml` function
* The `dtype` has checked in `quantize` function

Reply:
The guideline is always to check and report error to the user as early as possible